src/import/ensureConfiguration.ts

```diff
@@ -0,0 +1,596 @@
+  const isFromScratch = packageJSON === DEFAULT_PACKAGE_JSON;
+  if (isFromScratch && !allowFromScratch) {
+    return;
+  }
+
+  // Create the new YAML document
+  const configPath = path.join(root, 'config.yaml');
+  const doc = new Document();
+  const contents = new CommentableYAMLMap();
+  // @ts-ignore See: https://github.com/eemeli/yaml/issues/156
+  doc.contents = contents;
+
+  // Set the comments for the document header. Comments differ depending on which
+  // files were used in generation (if any).
+  let docComments: string;
+  if (isFromScratch) {
+    docComments =
+      'This SUSHI config.yaml file is provided as an example. You must modify its values to ' +
+      'accurately reflect the details of your project.';
+  } else {
+    docComments =
+      'This SUSHI config.yaml file was generated from the following configuration file(s):\n';
+    if (packageJSON !== DEFAULT_PACKAGE_JSON) {
+      docComments += '* package.json\n';
+    }
+    if (igIni !== DEFAULT_IG_INI) {
+      docComments += `* ${path.join('ig-data', 'ig.ini')}\n`;
+    }
+    if (packageList !== DEFAULT_PACKAGE_LIST) {
+      docComments += `* ${path.join('ig-data', 'package-list.json')}\n`;
+    }
+    docComments +=
+      'Please review this new config.yaml file and, make edits as necessary. This file should be considered the ' +
+      'new gold source configuration for SUSHI. Since this is the new gold source, you should delete the files ' +
+      'listed above since they are now unnecessary and redundant with this file.';
+  }
+  docComments +=
+    '\n\nThis new configuration is based on ImplementationGuide and other FHIR configuration formats but provides ' +
+    'additional user-friendly defaults and representations of information. This format is experimental and subject ' +
+    'to change. Your feedback is crucial. For more information about the new configuration format, please see the ' +
+    'SUSHI documentation at: https://build.fhir.org/ig/HL7/fhir-shorthand/sushi.html';
+  doc.commentBefore = wrap(docComments);
+
+  // id
+  const id = packageJSON.name ?? DEFAULT_PACKAGE_JSON.name;
+  contents.addPair('id', id);
+  // canonical
+  const canonical = packageJSON.canonical ?? DEFAULT_PACKAGE_JSON.canonical;
+  contents.addPair('canonical', canonical);
+  // url
+  contents.addCommentedOutPair(
+    'url',
+    `${canonical}/ImplementationGuide/${id}`,
+    'NOTE: url defaults to a generated value based on canonical and id. Uncomment to set your own.'
+  );
+  // version
+  contents.addPair('version', packageJSON.version ?? igIni.version ?? DEFAULT_PACKAGE_JSON.version);
+  // name (mimics old IG Exporter algorithm)
+  const name = (packageJSON.title ?? packageJSON.name ?? DEFAULT_PACKAGE_JSON.title).replace(
+    /[^A-Za-z0-9_]/g,
+    ''
+  );
+  contents.addPair('name', name);
+  // title (mimics old IG Exporter algorithm)
+  contents.addPair('title', packageJSON.title ?? packageJSON.name);
```

Comment:
Yeah, we can do that -- although if their `packageJSON` doesn't have at least a `name` then that would have been a pretty big problem for them in the past (since `name` is the id for the package).

Reply:
As of e661b0b, if I can't generate the title from existing config, I exclude it.